apps/web-giddh/src/app/new-inventory/new-inventory.component.ts

```diff
@@ -0,0 +1,194 @@
+        }
+    }
+    /* show/hide funcation search input field */
+    public searhcGroup(){
+        this.inputFullWidth = !this.inputFullWidth
+    }
+    public focusOnInput(){
+
+    }
+    public expandDateRang(){
+        this.dateRangFullWidth = !this.dateRangFullWidth
+    }
+
+
+    /* Aside pane toggle fixed class */
+    public toggleBodyClass(): void {
+        if (this.asideMenuState === 'in') {
+            document.querySelector('body').classList.add('fixed');
+        } else {
+            document.querySelector('body').classList.remove('fixed');
+        }
+    }
+
+    /* Aside pane open function */
+    public toggleAsidePane(event?): void {
+        if (event) {
+            event.preventDefault();
+        }
+        this.asideMenuState = this.asideMenuState === 'out' ? 'in' : 'out';
+        this.toggleBodyClass();
+    }
+
+    /* Create group aside pane open function */
+    public createGroupToggleAsidePane(event?): void {
+        if (event) {
+            event.preventDefault();
+        }
+        this.asideMenuState = this.asideMenuState === 'out' ? 'in' : 'out';
+        this.toggleBodyClass();
+    }
+
+    /* Create item aside pane open function */
+    public createItemToggleAsidePane(event?): void {
+        if (event) {
+            event.preventDefault();
+        }
+        this.asideMenuState = this.asideMenuState === 'out' ? 'in' : 'out';
+        this.toggleBodyClass();
+    }
+
+    /* Create unit aside pane open function */
+    public createUnitToggleAsidePane(event?): void {
+        if (event) {
+            event.preventDefault();
+        }
+        this.asideMenuState = this.asideMenuState === 'out' ? 'in' : 'out';
+        this.toggleBodyClass();
+    }
+
+    /* Create combo aside pane open function */
+    public createComboToggleAsidePane(event?): void {
+        if (event) {
+            event.preventDefault();
+        }
+        this.asideMenuState = this.asideMenuState === 'out' ? 'in' : 'out';
+        this.toggleBodyClass();
+    }
+
+    /**
+    * This will toggle the settings popup
+    *
+    * @param {*} [event]
+    * @memberof SettingsComponent
+    */
+    public toggleSettingPane(event?): void {
+        this.toggleBodyClass();
+
+        if (this.isMobileScreen && event && this.asideInventorySidebarMenuState === 'in') {
+            this.asideInventorySidebarMenuState = "out";
+        }
+    }
+
+    constructor(
+        private generalService: GeneralService,
+        private modalService: BsModalService,
+        private _breakPointObservar: BreakpointObserver
+    ) { }
+
+    /* advance serach modal */
+    openModal(inventoryAdvanceSearch: TemplateRef<any>) {
+        this.modalRef = this.modalService.show(inventoryAdvanceSearch,
+            Object.assign({}, { class: 'modal-lg' })
+        );
+    }
+
+    public ngOnInit() {
+        this._breakPointObservar.observe([
+            '(max-width: 767px)'
+        ]).pipe(takeUntil(this.destroyed$)).subscribe(result => {
```

Comment:
this.destroyed$ is not implemented in ngOnDestroy

Reply:
Done